pkg/finality-grandpa/voter_test.go

```diff
@@ -0,0 +1,703 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVoter_TalkingToMyself(t *testing.T) {
+	var localID ID = 5
+	voters := NewVoterSet([]IDWeight[ID]{
+		{localID, 100},
+	})
+
+	network := NewNetwork()
+
+	env := newEnvironment(network, localID)
+
+	// initialize chain
+	var lastFinalized HashNumber[string, uint32]
+	env.WithChain(func(chain *dummyChain) {
+		chain.PushBlocks(GenesisHash, []string{"A", "B", "C", "D", "E"})
+		lastFinalized.Hash, lastFinalized.Number = chain.LastFinalized()
+	})
+
+	finalized := env.FinalizedStream()
+	voter, globalOut := NewVoter[string, uint32, Signature, ID](
+		&env,
+		*voters,
+		make(chan globalInItem),
+		0,
+		nil,
+		lastFinalized,
+		lastFinalized,
+	)
+
+	globalIn := network.MakeGlobalComms(globalOut)
+	voter.globalIn = newWakerChan(globalIn)
+
+	done := make(chan any)
+	go func() {
+		defer close(done)
```

Comment:
why not just use a waitgroup instead of this. That would be more obvious while reading.

Reply:
I find `done` channels semantically correct when I'm only waiting on one goroutine.

Reply:
This stackoverflow question explains it in better way than I could 
https://stackoverflow.com/questions/36056615/what-is-the-advantage-of-sync-waitgroup-over-channels

Reply:
That post makes a lot of sense to me in the situation of waiting on multiple goroutines. However, I think what Tim is getting at (and I would say I agree) is that there isn't a need to use waitgroups over channels in this case because there is no "group" to wait on

Reply:
I agree.  I do use waitgroups in the tests when I'm waiting on more than 1 goroutine.